lib/matplotlib/image.py

```diff
@@ -587,94 +580,40 @@ def __init__(self, ax,
-                                filternorm=filternorm,
-                                filterrad=filterrad,
-                                resample=resample,
-                                **kwargs
-                                )
+        _ImageBase.__init__(self, ax,
+                            cmap=cmap,
+                            norm=norm,
+                            interpolation=interpolation,
+                            origin=origin,
+                            filternorm=filternorm,
+                            filterrad=filterrad,
+                            resample=resample,
+                            **kwargs
+        )
 
     def get_window_extent(self, renderer=None):
         x0, x1, y0, y1 = self._extent
         bbox = Bbox.from_extents([x0, y0, x1, y1])
         return bbox.transformed(self.axes.transData)
 
-    def make_image(self, magnification=1.0):
-        if self._A is None:
-            raise RuntimeError('You must first set the image'
-                               ' array or the image attribute')
-
+    def make_image(self, renderer, magnification=1.0, unsampled=False):
+        trans = self.get_transform()
         # image is created in the canvas coordinate.
         x1, x2, y1, y2 = self.get_extent()
-        trans = self.get_transform()
-        xy = trans.transform(np.array([(x1, y1),
-                                       (x2, y2),
-                                       ]))
-        _x1, _y1 = xy[0]
-        _x2, _y2 = xy[1]
-
-        transformed_viewLim = mtransforms.TransformedBbox(self.axes.viewLim,
-                                                          trans)
-
-        im, xmin, ymin, dxintv, dyintv, sx, sy = \
-            self._get_unsampled_image(self._A, [_x1, _x2, _y1, _y2],
-                                      transformed_viewLim)
-
-        fc = self.axes.patch.get_facecolor()
-        bg = mcolors.colorConverter.to_rgba(fc, 0)
-        im.set_bg(*bg)
-
-        # image input dimensions
-        im.reset_matrix()
-        numrows, numcols = im.get_size()
-        if numrows < 1 or numcols < 1:   # out of range
-            return None
-        im.set_interpolation(_interpd_[self._interpolation])
+        bbox = Bbox(np.array([[x1, y1], [x2, y2]]))
+        transformed_bbox = TransformedBbox(bbox, trans)
 
-        im.set_resample(self._resample)
-
-        # the viewport translation
-        if dxintv == 0.0:
-            tx = 0.0
-        else:
-            tx = (xmin-transformed_viewLim.x0)/dxintv * numcols
-        if dyintv == 0.0:
-            ty = 0.0
-        else:
-            ty = (ymin-transformed_viewLim.y0)/dyintv * numrows
-
-        im.apply_translation(tx, ty)
-
-        l, b, r, t = self.axes.bbox.extents
-        widthDisplay = ((np.round(r*magnification) + 0.5) -
-                        (np.round(l*magnification) - 0.5))
-        heightDisplay = ((np.round(t*magnification) + 0.5) -
-                         (np.round(b*magnification) - 0.5))
```

Comment:
That's right.  On close inspection of the test images (and some of the doc examples), I determined the old behavior is wrong -- it could leave gaps between the image and the spines at certain scales/dpis.